addons/website_sale_loyalty/models/loyalty_program.py

```diff
@@ -1,13 +1,23 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import fields, models
+from odoo import api, fields, models
 
 
 class LoyaltyProgram(models.Model):
     _name = 'loyalty.program'
     _inherit = ['loyalty.program', 'website.multi.mixin']
 
     ecommerce_ok = fields.Boolean("Available on Website", default=True)
+    hide_trigger_product_warning = fields.Boolean(compute="_compute_hide_trigger_product_warning")
```

Comment:
```suggestion
    show_trigger_product_warning = fields.Boolean(compute="_compute_hide_trigger_product_warning")
```
Or even, for clarity:
```suggestion
    show_non_published_product_warning = fields.Boolean(compute="_compute_hide_trigger_product_warning")
```
My reasoning is that the default value is False, "showing" is the action. Additionally, we try to keep var names positive, but that's more a personal taste.